data/typescript/lib/net.ts

```diff
@@ -20,8 +20,29 @@ class Server extends EventEmitter {
 
     listen(port: number, callback: () => void) {
         var h = this._createHandle();
+        var z = null;  // Make things fun for PTA
+
+        // TODO: uncomment and analyze
+        // function interesting1(a: number) {
+        //     var z = this._createHandle();
+        //     return 2;
+        // }
+        //
+        // var interesting2 = () => {
+        //     z = interesting1(3);
+        //     return 3;
+        // };
+        //
+        // var x = interesting2();
+
         try {
             h.bind(port); h.listen(5);
+
+            // And to make them even more interesting :-)
+            z = h;
+            if (z == 42) {
+                return 15;
+            }
```

Comment:
I suggest we put these snippets into a separate test file, and then we can make it into a small test suite.

Reply:
ok, I'll create a separate file